examples/mksomersaultmodifiedpdx.py

```diff
@@ -0,0 +1,230 @@
+somersault_dlr = dlc.base_variants.somersault.diag_layer_raw
+
+# rename the "session_start" and "session_stop" services to
+# "start_session" and "stop_session"
+start_service = find_named_object(somersault_dlr.diag_comms, "session_start")
+assert isinstance(start_service, DiagService)
+start_service.short_name = "start_session"
+
+stop_service = find_named_object(somersault_dlr.diag_comms, "session_stop")
+assert isinstance(stop_service, DiagService)
+stop_service.short_name = "stop_session"
+
+tester_present_service = find_named_object(somersault_dlr.diag_comms, "tester_present")
+assert isinstance(tester_present_service, DiagService)
+tester_present_pr = tester_present_service.positive_responses[0]
+param = tester_present_pr.parameters.status
+assert isinstance(param, CodedConstParameter)
+param.diag_coded_type = somersaultecu.somersault_diagcodedtypes["uint16"]
+param.coded_value = 0x1234
+
+# change the DOP used by the "can_do_backward_flips" parameter of the
+# positive response to the "session" service from "bool" to "uint8"
+cdbf_param = start_service.positive_responses[0].parameters.can_do_backward_flips
+doc_frags = start_service.odx_id.doc_fragments
+assert isinstance(cdbf_param, ValueParameter)
+cdbf_param.dop_ref = OdxLinkRef("somersault.DOP.uint8", doc_frags)
+
+# change the byte and bit positions of the "num_flips_done" parameter
+# of the "grudging_backward" response for all variants variant
+gb_response = find_named_object(somersault_dlr.positive_responses, "grudging_backward")
+param = gb_response.parameters.num_flips_done
+param.byte_position = 2
+param.bit_position = 4
+
+# add a new "somersault_young" variant which can do flic-flacs and
+# does not take any instructions
+somersault_young = deepcopy(dlc.ecu_variants.somersault_lazy)
+somersault_young_dlr = dlc.ecu_variants.somersault_lazy.diag_layer_raw
+somersault_young_dlr.short_name = "somersault_young"
+somersault_young_dlr.odx_id = OdxLinkId("ECU.somersault_young",
+                                        somersault_young_dlr.odx_id.doc_fragments)
+somersault_young_dlr.description = \
+"""<p>A young version of the somersault ECU
+
+It is as grumpy as the lazy variant, but it is more agile, so it can do flic-flacs.
+
+On the flipside, it is unwilling to take any instructions, so no
+operational parameters can be set. Finally, it is unwilling to compete
+(i.e. it does not time its somersaults).</p>"""
+
+# remove the "sault_time" parameter from the positive response of the
+# "do_forward_flips" service.
+do_forward_flips_service = somersault_young.services.do_forward_flips
+pr = do_forward_flips_service.positive_responses[0]
+new_params = [x for x in pr.parameters if getattr(x, "short_name", None) != "sault_time"]
+pr.parameters = NamedItemList(new_params)
+
+# add a "flic-flac" service
+flic_flac_request = Request(
+    odx_id=OdxLinkId("somersault.RQ.flic_flac", doc_frags),
+    short_name="RQ_flic_flac",
+    long_name=None,
+    description=None,
+    admin_data=None,
```

Comment:
we need a better object creation api, that don't require providing optional properties

Reply:
I agree. the problem is that using dataclasses with defaults does not want play with inheritance...

Reply:
How so?

Reply:
Migrate to 3.10 and use `kw_only`

Reply:
I opened an issue to discuss this and to keep track of stuff which we should change after the transition: https://github.com/mercedes-benz/odxtools/issues/286